### PR TITLE
Change Simple node scripts to instantiate themselves

### PR DESCRIPTION
**Pull request by dylanmccall**
To support users adding SimpleCharacter and SimpleScoring as nodes, instead of as child scenes, add an _init function for each which loads the appropriate scene file and instantiates it.

https://phabricator.endlessm.com/T35545

### CODE REVIEW

File: addons/block_code/simple_nodes/simple_scoring/simple_scoring.gd [hunk 1]
```diff
@@ -19,6 +19,13 @@ const _POSITIONS_FOR_PLAYER = {
 }
 
 
+func _init():
+	var node = preload("res://addons/block_code/simple_nodes/simple_scoring/_simple_scoring.tscn").instantiate() as Node
+	node.replace_by(self, true)
```
Comment (starnight) [inline addons/block_code/simple_nodes/simple_scoring/simple_scoring.gd:24]: When I play the Pong game, it shows error:
```
E 0:00:00:0452   simple_scoring.gd:24 @ _init(): Condition "p_node->data.parent" is true.
  <C++ Source>   scene/main/node.cpp:2796 @ replace_by()
  <Stack Trace>  simple_scoring.gd:24 @ _init()
                 block_code.gd:56 @ _update_parent_script()
                 block_code.gd:14 @ _ready()
```

File: addons/block_code/simple_nodes/simple_character/simple_character.gd [hunk 2]
```diff
@@ -52,6 +52,13 @@ func on_gravity_changed(new_gravity):
 	gravity = new_gravity
 
 
+func _init():
+	var node = preload("res://addons/block_code/simple_nodes/simple_character/_simple_character.tscn").instantiate() as Node
+	node.replace_by(self, true)
```
Comment (starnight) [inline addons/block_code/simple_nodes/simple_character/simple_character.gd:57]: When I play the Pong game, it shows error:
```
E 0:00:00:0445   simple_character.gd:57 @ _init(): Condition "p_node->data.parent" is true.
  <C++ Source>   scene/main/node.cpp:2796 @ replace_by()
  <Stack Trace>  simple_character.gd:57 @ _init()
                 block_code.gd:56 @ _update_parent_script()
                 block_code.gd:14 @ _ready()
```

**Review by starnight — CHANGES_REQUESTED** [review]

File: addons/block_code/simple_nodes/simple_scoring/simple_scoring.gd (hunk 1 above)
Comment (dylanmccall) [reply addons/block_code/simple_nodes/simple_scoring/simple_scoring.gd:27]: Ooh, I see how that happened. That's fixed in the newest set of commits.

File: addons/block_code/simple_nodes/simple_character/_simple_character.gd [hunk 3]
```diff
@@ -0,0 +1,8 @@
+@tool
+extends SimpleCharacter
+
+
+func _init():
+	# Disable the SimpleCharacter _init function. We expect this script is
+	# already bound to _simple_character.tscn.
+	pass
```
Comment (manuq) [inline addons/block_code/simple_nodes/simple_character/_simple_character.gd:8]: :+1:

**Review by manuq — APPROVED** [review]

**Pull request merged**